packages/grafana-data/src/datetime/datemath.ts

```diff
@@ -40,12 +40,10 @@ export function parse(
   }
 
   if (typeof text !== 'string') {
-    if (isDateTime(text)) {
-      return text;
-    }
-    if (isDate(text)) {
-      return dateTime(text);
+    if (isDateTime(text) || isDate(text)) {
+      return dateTimeForTimeZone(timezone, text);
```

Comment:
if I understand correctly, before the `parse` function ignored the `timezone` argument, and returned a `moment` date with utc. 

>  * @param timezone Only string 'utc' is acceptable here, for anything else, local timezone is used.

It seems this was on purpose. If we change that behavior, it means that we can get side effects of people relying on expecting UTC, right? I wonder if we can tackle this issue differently 🤔

For instance, we could modify the `parseUrlParam` to handle `timezone`, instead of directly converting to UTC,  we could parse the epoch time and convert it to the specified timezone? 

```
 if (!isNaN(Number(value))) { 
   const epoch = parseInt(value, 10); 
    return moment.tz(epoch, timezone);
 } 

```